shoop/core/pricing/priceful.py

```diff
@@ -0,0 +1,207 @@
+
+        :rtype: decimal.Decimal
+        """
+        if not self.base_price:
+            return decimal.Decimal(0)
+        return 1 - (self.price / self.base_price)
+
+    @property
+    def discount_percentage(self):
+        """
+        Discount percentage, 100 meaning totally discounted.
+
+        See `discount_rate`.
+
+        :rtype: decimal.Decimal
+        """
+        return self.discount_rate * 100
+
+    @property
+    def is_discounted(self):
+        """
+        Check if there is a discount in effect.
+
+        :return: True, iff price < base price.
+        """
+        return (self.price < self.base_price)
+
+    @property
+    def discounted_unit_price(self):
+        """
+        Unit price with discount.
+
+        If quantity is 0, will return ``base_unit_price - discount_amount``.
+
+        :rtype: shoop.core.pricing.Price
+        """
+        return self.base_unit_price - (self.discount_amount / (self.quantity or 1))
+
+    @property
+    def unit_discount_amount(self):
+        """
+        Discount amount per unit.
+
+        If quantity is 0, will return ``discount_amount``.
+
+        :rtype: shoop.core.pricing.Price
+        """
+        return self.discount_amount / (self.quantity or 1)
+
+    @property
+    def tax_rate(self):
+        """
+        :rtype: decimal.Decimal
+        """
+        taxless = self.taxless_price
+        taxful = self.taxful_price
+        if not taxless.amount:
+            return decimal.Decimal(0)
+        return (taxful.amount / taxless.amount) - 1
+
+    @property
+    def tax_percentage(self):
+        """
+        :rtype: decimal.Decimal
+        """
+        return self.tax_rate * 100
+
+    @property
+    def taxful_price(self):
+        """
+        :rtype: TaxfulPrice
+        """
+        price = self.price
+        if price.includes_tax:
+            return price
+        else:
+            return TaxfulPrice(price.amount + self.tax_amount)
+
+    @property
+    def taxless_price(self):
+        """
+        :rtype: TaxlessPrice
+        """
+        price = self.price
+        if price.includes_tax:
+            return TaxlessPrice(price.amount - self.tax_amount)
+        else:
+            return price
+
+    taxful_base_price = TaxfulFrom('base_price')
+    taxless_base_price = TaxlessFrom('base_price')
+
+    taxful_discount_amount = TaxfulFrom('discount_amount')
+    taxless_discount_amount = TaxlessFrom('discount_amount')
+
+    taxful_base_unit_price = TaxfulFrom('base_unit_price')
+    taxless_base_unit_price = TaxlessFrom('base_unit_price')
+
+    taxful_discounted_unit_price = TaxfulFrom('discounted_unit_price')
+    taxless_discounted_unit_price = TaxlessFrom('discounted_unit_price')
+
+    taxful_unit_discount_amount = TaxfulFrom('unit_discount_amount')
+    taxless_unit_discount_amount = TaxlessFrom('unit_discount_amount')
+
+    # TODO: Replace occurences of these aliases with their source fields
+
+    total_price = Alias('price')
```

Comment:
Sure. Just figured that'd be more automatic in a sense. But I guess we can add the warning feature during that cleanup pass.